### PR TITLE
[MISC] Adds new concept template_specialisation_of.

### PR DESCRIPTION
**Pull request by Irallia**
Resolves partly https://github.com/seqan/product_backlog/issues/259

### CODE REVIEW

**Comment by marehr** [comment]
Did you talk with @smehringer about this? Or did you pick this yourself?

**Comment by Irallia** [comment]
> 
> 
> Did you talk with @smehringer about this? Or did you pick this yourself?

She gave me this issue.

**Comment by marehr** [comment]
I guess this is blocked by #2265. We should first do the move of `{core/type_traits/ => core/detail/}template_inspection.hpp`

See https://gist.github.com/rrahn/f6b09ad67bebd14bd9c58f77c53e0450

**Review by marehr — COMMENTED** [review]
Postponed until the header template_inspection.hpp was moved.

**Review by marehr — APPROVED** [review]
thank you

File: include/seqan3/core/detail/template_inspection.hpp [hunk 1]
```diff
@@ -235,6 +236,36 @@ struct valid_template_spec_or<fallback_t, templ_t, spec_t...>
...
+ * \{
+ */
+
+/*!\interface seqan3::template_specialisation_of <>
+ * \brief Provides concept `seqan3::template_specialisation_of<mytype, [...]>` for checking the type specialisation of
+ *        some type with a given template, for example a specialized `type_list<float>` with the `type_list` template.
+ *
+ * \ingroup type_traits
+ *
+ * \tparam mytype           The source type.
```
Comment (smehringer) [inline include/seqan3/core/detail/template_inspection.hpp:253]:
```suggestion
 * \tparam mytype           The guery type.
```

? 
Source type somehow threw me off. Maybe query isn't the best alternative thought. `The type to check` ?
Source type sounds like I'm actually doing something with it or taking something from it to put somewhere else..

File: test/snippet/core/detail/template_inspection_usage_3.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#include <vector>
+
+#include <seqan3/core/detail/template_inspection.hpp>
+
+int main()
+{
+    using my_type = std::vector<int>;
+
+    if constexpr (seqan3::detail::template_specialisation_of<my_type, std::vector>) // std::vector has no <> !
```
Comment (smehringer) [inline test/snippet/core/detail/template_inspection_usage_3.cpp:9]:
```suggestion
    if constexpr (seqan3::detail::template_specialisation_of<my_type, std::vector>) // Note: std::vector has no <> !
```

**Review by smehringer — APPROVED** [review]
Just to tiny documentation things. After including them this can be merged.

**Pull request merged**